### PR TITLE
Fixed creeper not exploding when 1-block higher than player

### LINKED ISSUE #708: Creepers don't explode when a player is within range but vertically one below
Being in a range of 1 horizontally but 1 vertically below the creeper will make him charge completly but he will never explode.

### PR DESCRIPTION
**Pull request by SafwatHalaby**
Fixes #708

### CODE REVIEW

File: src/Mobs/AggressiveMonster.cpp [hunk 1]
```diff
@@ -76,9 +76,11 @@ void cAggressiveMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 	}
 
 	cTracer LineOfSight(GetWorld());
-	Vector3d AttackDirection(m_Target->GetPosition() - GetPosition());
+	Vector3d MyHeadPosition = GetPosition() + Vector3d(0, GetHeight(), 0);
+	Vector3d AttackDirection(m_Target->GetPosition() + Vector3d(0, m_Target->GetHeight(), 0) - MyHeadPosition);
 
-	if (ReachedFinalDestination() && !LineOfSight.Trace(GetPosition(), AttackDirection, (int)AttackDirection.Length()))
+
+	if (ReachedFinalDestination() && !LineOfSight.Trace(MyHeadPosition, AttackDirection, (int)AttackDirection.Length()))
```
Comment (bearbin) [inline src/Mobs/AggressiveMonster.cpp]: static_cast?

**Comment by SafwatHalaby** [comment]
Squashed. But do not merge yet, I want to test it a bit more.

File: src/Mobs/AggressiveMonster.cpp (hunk 1 above)
Comment (SafwatHalaby) [inline src/Mobs/AggressiveMonster.cpp]: Done.

File: MCServer/Plugins/Core [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 5171b43807ff699a6b239ad4969520730b3748a2
+Subproject commit ea0ab964d568630fd4f2b52954186f2851a769e1
```
Comment (bearbin) [inline MCServer/Plugins/Core]: You seem to have committed a downgrade to Core here.

Comment (SafwatHalaby) [inline MCServer/Plugins/Core]: Fixed and squashed.

**Comment by SafwatHalaby** [comment]
Testing done, merge if compiles.

**Pull request merged**